tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -27,21 +34,81 @@ public void run() {
     @Override
     public void process(final Socket connection) {
         try (final var inputStream = connection.getInputStream();
+             final var bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
              final var outputStream = connection.getOutputStream()) {
+            String requestStartLine = bufferedReader.readLine();
+            HttpRequestUrl url = new HttpRequestUrl(requestStartLine.split(" ")[1]);
 
-            final var responseBody = "Hello world!";
-
-            final var response = String.join("\r\n",
-                    "HTTP/1.1 200 OK ",
-                    "Content-Type: text/html;charset=utf-8 ",
-                    "Content-Length: " + responseBody.getBytes().length + " ",
-                    "",
-                    responseBody);
+            String response;
+            if (url.equalPath("/")) {
+                response = create200Response("Hello world!", ContentType.TEXT_PLAIN);
+            } else if (url.equalPath("/login")) {
+                String account = url.getParameter("account");
+                String password = url.getParameter("password");
+                if (account != null && password != null) {
+                    validateAccount(account, password);
+                }
+                response = createStaticResourceResponse("/login.html");
+            } else if (url.isStaticResourcePath()) {
+                response = createStaticResourceResponse(url.getPath());
+            } else {
+                response = create404Response();
+            }
 
             outputStream.write(response.getBytes());
             outputStream.flush();
         } catch (IOException | UncheckedServletException e) {
             log.error(e.getMessage(), e);
         }
     }
+
+    private String createStaticResourceResponse(String path) throws IOException {
+        Path staticResource = getStaticResource(path);
+        if (staticResource == null) {
+            return create404Response();
+        }
+        String fileExtension = path.split("\\.")[1];
+        return create200Response(Files.readString(staticResource), ContentType.of(fileExtension));
+    }
+
+    private void validateAccount(String account, String password) {
+        Optional<User> optionalUser = InMemoryUserRepository.findByAccount(account);
+        if (optionalUser.isEmpty()) {
+            log.info("존재하지 않는 유저입니다.");
+        } else {
+            User user = optionalUser.get();
+            if (user.checkPassword(password)) {
+                log.info(user.toString());
+            } else {
+                log.info("비밀번호가 일치하지 않습니다.");
+            }
+        }
+    }
```

Comment:
여기도 if문이 중첩되어있는데, 가독성 향상을 위해 early return 사용해보시는거 어떤가용??

```suggestion
private void validateAccount(String account, String password) {
    Optional<User> optionalUser = InMemoryUserRepository.findByAccount(account);
    if (optionalUser.isEmpty()) {
        log.info("존재하지 않는 유저입니다.");
        return; 
    }

    User user = optionalUser.get();
    if (!user.checkPassword(password)) {
        log.info("비밀번호가 일치하지 않습니다.");
        return;
    }

    log.info(user.toString());
}
```

Reply:
마찬가지로 반영했습니다!!
[refactor: 가독성 개선을 위한 early return 적용](https://github.com/woowacourse/java-http/pull/831/commits/2c085dbc3c4883d90e600f519188a9f05dc5f3e7)